src/examples/solana_stake.ts

```diff
@@ -0,0 +1,97 @@
+
+import { StakingReward } from "../coinbase/staking_reward";
+import { Coinbase } from "../coinbase/coinbase";
+import { Address } from "../coinbase/address";
+import { NetworkIdentifier, StakingOperation } from "../client";
+import { coinbaseApiKeyPath, getKeypair } from "./solana_wallet";
+import { KeyPairSigner, createSolanaRpc, decodeTransactionMessage, getBase58Codec, getBase58Decoder, getBase64EncodedWireTransaction, getBase64Encoder, getCompiledTransactionMessageCodec, getSignersFromTransactionMessage, getTransactionDecoder, sendAndConfirmTransactionFactory, signAndSendTransactionMessageWithSigners, signTransaction } from "@solana/web3.js";
+import { TextEncoder } from "util";
+import * as bs58 from 'bs58';
+
+// import { StakingOperation } from "../coinbase/staking_operation";
+
+async function stakeOperations(signer: KeyPairSigner) {
+  const ctx = (await Coinbase.apiClients.stake?.getStakingContext({ network_id: NetworkIdentifier.SolanaDevnet, address_id: signer.address.toString(), asset_id: Coinbase.assets.Sol, options: {} }))?.data.context;
+
+  console.log("Stakeable: ");
+  console.log(ctx?.stakeable_balance.amount);
+  console.log(ctx?.stakeable_balance.asset);
+  let ops: Array<StakingOperation> = new Array<StakingOperation>();
+
+  {
+    const amount = ctx?.stakeable_balance.amount ? BigInt(ctx?.stakeable_balance.amount.toString()) : 0;
+    if (amount > 1)
+      ops.push((await Coinbase.apiClients.stake!.buildStakingOperation({
+        action: "stake",
+        address_id: signer.address.toString(),
+        asset_id: Coinbase.assets.Sol,
+        network_id: NetworkIdentifier.SolanaDevnet,
+        options: { amount: "100000000" },
+      }))!.data!);
+  }
+
+  console.log("Unstakeable: ");
+  console.log(ctx?.unstakeable_balance.amount);
+  console.log(ctx?.unstakeable_balance.asset);
+
+  {
+    const amount = ctx?.unstakeable_balance.amount ? BigInt(ctx?.unstakeable_balance.amount.toString()) : 0;
```

Comment:
Seeing a lot of optional chaining in these examples. Is this OK for examples, or should we attempt to remove these for clarity?

Reply:
This is just for output formatting, I think it is fine